modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -52,6 +61,16 @@
     private boolean isSlotActive;
 
     /**
+     * Indicates whether the slot is a fresh one or an overlapped one
+     */
+    private boolean isAnOverlappingSlot;
```

Comment:
Yes, that should work given that only overlapped slots are given repetitively. Will check on it. Good suggestion.